modules/apps/frontend-data-set/frontend-data-set-admin-api/src/main/java/com/liferay/frontend/data/set/admin/model/DataSetModelManager.java

```diff
@@ -0,0 +1,15 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.data.set.admin.model;
+
+/**
+ * @author Daniel Sanz
+ */
+public interface DataSetModelManager {
+
+	public void checkCompany(long companyId);
```

Comment:
Does this need an interface? Will there ever be a different implementation of data set persistence generation? Maybe this could be a `*Manager` class in a public package in `frontend-data-set-admin-web`.

Reply:
I had to use an interface, with its corresponding `-api` module because of SF restrictions:

- SF does not like a @Component to declare a `service` equals to its own class (`DataSetModelManagerImpl`)  so an interface is needed
- Interface can't live in the -web package even if we add it to the `Export-Package` bnd header
- if I just use an anonymous `@Component()`, that is, one with does nmo include `service` then the dependencies can't be resolved from the `CompanyModelListener` and OSGi complains about this
- But still we need the `activate()` method for this to work.

If we manage to find a different way, I'll be glad to remove the interface. The only way I see now is to tie the `*Manager`  to the portlet's `activate()` method.

Reply:
> SF does not like a https://github.com/component to declare a service equals to its own class (DataSetModelManagerImpl) so an interface is needed

This seems odd. What about something like [TableReferenceDefinitionManager](https://github.com/liferay/liferay-portal/blob/master/modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/reference/TableReferenceDefinitionManager.java). It's a class with no interfaces, a @Component with service = own name, and @Activate method.

Reply:
Oh, above example is in `-service` module. Is that the problem? Should the new module name be `frontend-data-set-admin-service`? If it has to be `-api`, implementation should be in matching `-impl`.

Reply:
for now the impl lives in `-web` but it can be easily moved to other module if that matches a pattern